Yoast/javascript
# Update store on image select

## Summary


This PR can be summarized in the following changelog entry:

* [social-metadata-previews] Fixed a bug where the TwitterImage did not update on selecting a new image.
* [social-metadata-previews] Fixed a bug where the FacebookImage did not correctly switch the mode of its container to square/landscape format.

## Relevant technical choices:

* The way information currently flows is a bit spaghetti, but we decided to not fix all the technical debt in this PR. This PR only fixes the lack of updating on prop changes.
* The social previews constantly mount and unmount upon closing the collapsibles / switching tabs. In the future it may be nicer to keep image state in the Redux store, in stead of re-calculating all the time. Again, not fixing in this PR to help the process along.
* **The persisting of the settings is still not functional and will be fixed in a different issue**.

## Test instructions

This PR can be tested by following these steps:

* To reproduce the bug: go to `wordpress-seo-premium`, try to select an image for Twitter, and note that it does not trigger anything in the image.
Similarly, select a square image for Facebook (just google a thumbnail or something), and note that the container stays in landscape.

* Link this branch to `wordpress-seo-premium`, and build all the things.
* Add an image to Twitter, note that it updates. Remove it again, note that it updates.
* Put the square image in Facebook, note that it is now also the square container. Put a landscape image, note that it is now landscape.
 

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2724
